ZeeHomeScreen/Classes/Components/BasicContainerCell/UniversalCollectionViewHeaderFooterView.m

```diff
@@ -5,27 +5,33 @@
 //  Created by Miri on 28/01/2020.
 //
 
-
 #import "UniversalCollectionViewHeaderFooterView.h"
 #import <ZeeHomeScreen/ZeeHomeScreen-Swift.h>
 
 @implementation UniversalCollectionViewHeaderFooterView
 
-
 -(void)prepareForReuse {
     [super prepareForReuse];
+
     _delegate = nil;
-    if ([self.componentViewController respondsToSelector:@selector(prepareComponentForReuse)]) {
-//        [self.componentViewController prepareComponentForReuse];
-    }
+
+    [self.componentViewController removeViewFromParentViewController];
+    _componentViewController = nil;
 }
 
 -(void)setComponentViewController:(UIViewController<ComponentProtocol> *)componentViewController {
+    if (_componentViewController != nil) {
+        [_componentViewController removeViewFromParentViewController];
```

Comment:
Release the previous view controller, when the new one comes in. It solves the problem with duplicating instances.